server/admin_management/api/app/routers/router_guards.py

```diff
@@ -121,71 +115,140 @@ async def get_target_user_from_id(
+                "user_name": rb_user_name,
+                "user_type_code": rb_user_type_code,
+                "user_guid": rb_user_guid
+            })
+            return target_new_user
+
+
+async def target_user_bceid_search(
+    requester: Requester = Depends(get_current_requester),
+    target_user: TargetUser = Depends(get_target_user_from_id)
+) -> TargetUser:
+    """
+    BCeID user search for target_user.
+    :param target_user "Depends(get_target_user_from_id)": the initial
+        dependency with "get_target_user_from_id" has initial target_user
+        parsed from the request. It's business_guid will be updated to the
+        target_user after user is searched through IDIM Proxy.
+    """
+    if (
+        target_user.user_type_code == UserType.BCEID and
+        (
+            target_user.is_new_user() or
+            target_user.business_guid is None
+        )
+    ):
+        user_guid = target_user.user_guid  # will be used by search using user_guid
+        LOGGER.debug(
+            f"Searching business guid for target_user: {target_user}, with requester {requester}"
+        )
+        idim_proxy_api = IdimProxyService(requester)
+        search_result = idim_proxy_api.search_business_bceid(
+            IdimProxyBceidSearchParam(**{
+                "searchUserBy": IdimSearchUserParamType.USER_GUID,
+                "searchValue": user_guid
+            })
+        )
+        if search_result.get("found"):
+            business_guid = search_result.get("businessGuid")
+            LOGGER.debug(
+                f"business_guid result: {business_guid} is updated to target_user."
             )
-            return TargetUser.model_validate(user) if user is not None else None
-        except json.JSONDecodeError:
-            return None
+            target_user.business_guid = business_guid
+        else:
+            LOGGER.debug(
+                f"IDIM look up user guid:`{user_guid}` returns no result. " +
+                "business_guid will NOT be updated to target_user."
+            )
+
+    return target_user
 
 
 async def enforce_self_grant_guard(
     requester: Requester = Depends(get_current_requester),
-    target_user: Union[TargetUser, None] = Depends(get_target_user_from_id),
+    target_user: TargetUser = Depends(get_target_user_from_id),
 ):
     """
     Verify logged on admin (requester):
         Self granting/removing privilege currently isn't allowed.
     """
     LOGGER.debug(f"enforce_self_grant_guard: requester - {requester}")
     LOGGER.debug(f"enforce_self_grant_guard: target_user - {target_user}")
-    if target_user is not None:
-        is_same_user_name = requester.user_name == target_user.user_name
-        is_same_user_type_code = requester.user_type_code == target_user.user_type_code
+    is_same_user_name = requester.user_name == target_user.user_name
+    is_same_user_type_code = requester.user_type_code == target_user.user_type_code
```

Comment:
also here since we have the user_guid for both target_user and requester, maybe we can user user_guid to replace "user_name and user_type_code"

Reply:
We haven't fully save user_guid for new user yet, I feel it can wait altogether when we do switch to use user_guid.